### PR TITLE
packaging: Automatically generate zulip_bots MANIFEST.in.

### LINKED ISSUE #69: zulip_bots: Autogenerate MANIFEST.in in setup().
Ideally, `zulip_bots/MANIFEST.in` should be automatically generated by `setup.py` and it should be ignored in `.gitignore`. There are a few things to keep in mind:

* We could add this as a part of a separate build step or command in `setup.py`.
* We need to make sure that this generation of `zulip_bots/MANIFEST.in` can be triggered from our main repo when installing the `zulip_bots` package, which might be a bit of a workaround. To see why this is a priority, see [this commit](https://github.com/zulip/python-zulip-api/pull/61/commits/8c8f682f61a0bc845eb9208ed935d946463765a3).

On a more general note, I feel like I'd like to get to a point where we have a script that completely automates the different release processes for our development repo vs. the actual package PyPA release, since PyPA packages shouldn't be shipped with fixtures and logos, but the repo should have them for the reasons described [here](https://github.com/zulip/python-zulip-api/pull/61/commits/8c8f682f61a0bc845eb9208ed935d946463765a3).

I'll probably pick this up eventually!

@timabbott: FYI

Thanks,
Eeshan

### PR DESCRIPTION
**Pull request by derAnfaenger**
Fixes #69.

I tested this by copying the requirements for this repo from the main repo, amending them to 
```
git+https://github.com/derAnfaenger/python-zulip-api.git@0.3.2#egg=zulip_bots==0.3.2&subdirectory=zulip_bots
git+https://github.com/derAnfaenger/python-zulip-api.git@0.3.2#egg=zulip==0.3.2&subdirectory=zulip
```
in a new file `requirements.txt`  and running `pip -r requirements.txt`, which passed.

Furthermore, I refactored the `generate_manifest.py` script and also made it work on Windows...

### CODE REVIEW

**Comment by timabbott** [comment]
@eeshangarg can you review this one?

**Comment by eeshangarg** [comment]
@timabbott: Sure! I'll take a look! :)

File: zulip_bots/setup.py [hunk 1]
```diff
@@ -5,6 +5,10 @@
 
 import sys
 
+# Before anything, generate MANIFEST.in
+import generate_manifest
+generate_manifest.main()
```
Comment (derAnfaenger) [inline zulip_bots/setup.py]: I don't know if this is pythonic.

File: zulip_bots/generate_manifest.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        fp.write('\n')
+
+        fp.writelines(logos)
+        fp.write('\n')
+
+        fp.writelines(docs)
+        fp.write('\n')
+
+if __name__ == '__main__':
+    main()
```
Comment (eeshangarg) [inline zulip_bots/generate_manifest.py]: Thanks for refactoring this! Your approach is more concise (and hence better) than mine! :)

File: zulip_bots/setup.py (hunk 1 above)
Comment (eeshangarg) [inline zulip_bots/setup.py]: I don't think its a big deal, honestly! Ideally, we would do this as part of a custom build step in `setup.py`. :)

File: zulip_bots/MANIFEST.in [hunk 3]
```diff
@@ -1,47 +1,47 @@
...
-include zulip_bots/bots/encrypt/doc.md
-include zulip_bots/bots/define/doc.md
-include zulip_bots/bots/helloworld/doc.md
 include zulip_bots/bots/tictactoe/doc.md
-include zulip_bots/bots/googlesearch/doc.md
+include zulip_bots/bots/virtual_fs/doc.md
+include zulip_bots/bots/weather/doc.md
 include zulip_bots/bots/xkcd/doc.md
+include zulip_bots/bots/yoda/doc.md
+include zulip_bots/bots/youtube/doc.md
```
Comment (eeshangarg) [inline zulip_bots/MANIFEST.in]: I'd like this file to eventually be added to `.gitignore`. Ideally, its not something we want future contributors to have to manually handle in any way before submitting a PR.

Comment (timabbott) [inline zulip_bots/MANIFEST.in]: That seems like a good step to do in a follow-up commit, because the diff is handy for checking correctness right now :)

**Comment by timabbott** [comment]
I went ahead and merged this; the outstanding issues all feel like things that would be better as a follow-up commit or series of commits.  I think next steps are:
* Move MANIFEST.in to no longer be checked in and instead be a generated file that lives in `.gitignore`
* Move the `setup.py` code to run using a more proper style.
* Decide about shipping the logos and test fixtures.  I actually am currently leaning towards the belief that the logos at least we will need to ship in the pypi unconditionally.

**Pull request closed without merging**